File: Sources/ReusableOnboarding/OnboardingTheme.swift
import SwiftUI

public struct OnboardingTheme {
    public let background: LinearGradient
    public let cardColor: Color
    public let raisedColor: Color
    public let highlightColor: Color
    public let primaryTextColor: Color
    public let secondaryTextColor: Color
    public let accentColor: Color
    public let warningColor: Color
    public let borderColor: Color
    public let buttonGradient: LinearGradient

    public init(
        background: LinearGradient,
        cardColor: Color,
        raisedColor: Color,
        highlightColor: Color,
        primaryTextColor: Color,
        secondaryTextColor: Color,
        accentColor: Color,
        warningColor: Color,
        borderColor: Color,
        buttonGradient: LinearGradient
    ) {
        self.background = background
        self.cardColor = cardColor
        self.raisedColor = raisedColor
        self.highlightColor = highlightColor
        self.primaryTextColor = primaryTextColor
        self.secondaryTextColor = secondaryTextColor
        self.accentColor = accentColor
        self.warningColor = warningColor
        self.borderColor = borderColor
        self.buttonGradient = buttonGradient
    }

    public static let standard = OnboardingTheme(
        background: LinearGradient(
            colors: [Color(hex: 0x05070D), Color(hex: 0x101B2D)],
            startPoint: .topLeading,
            endPoint: .bottomTrailing
        ),
        cardColor: Color(hex: 0x111827),
        raisedColor: Color(hex: 0x1F2937),
        highlightColor: Color(hex: 0x1D4ED8, alpha: 0.2),
        primaryTextColor: Color(hex: 0xF9FAFB),
        secondaryTextColor: Color(hex: 0xAAB2C0),
        accentColor: Color(hex: 0x2563EB),
        warningColor: Color(hex: 0xF59E0B),
        borderColor: Color.white.opacity(0.16),
        buttonGradient: LinearGradient(
            colors: [Color(hex: 0x2563EB), Color(hex: 0x1D4ED8)],
            startPoint: .topLeading,
            endPoint: .bottomTrailing
        )
    )
}

private extension Color {
    init(hex: UInt32, alpha: Double = 1.0) {
        let red = Double((hex >> 16) & 0xFF) / 255.0
        let green = Double((hex >> 8) & 0xFF) / 255.0
        let blue = Double(hex & 0xFF) / 255.0
        self.init(.sRGB, red: red, green: green, blue: blue, opacity: alpha)
    }
}

File: onboarding-android/src/main/kotlin/com/vadimkomis/onboarding/OnboardingTheme.kt
package com.vadimkomis.onboarding

import androidx.compose.runtime.Immutable
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color

@Immutable
public data class OnboardingTheme(
    public val background: Brush,
    public val cardColor: Color,
    public val raisedColor: Color,
    public val highlightColor: Color,
    public val primaryTextColor: Color,
    public val secondaryTextColor: Color,
    public val accentColor: Color,
    public val warningColor: Color,
    public val borderColor: Color,
    public val buttonGradient: Brush,
) {
    public companion object {
        public val standard: OnboardingTheme = OnboardingTheme(
            background = Brush.linearGradient(
                colors = listOf(Color(0xFF05070D), Color(0xFF101B2D)),
            ),
            cardColor = Color(0xFF111827),
            raisedColor = Color(0xFF1F2937),
            highlightColor = Color(0x331D4ED8),
            primaryTextColor = Color(0xFFF9FAFB),
            secondaryTextColor = Color(0xFFAAB2C0),
            accentColor = Color(0xFF2563EB),
            warningColor = Color(0xFFF59E0B),
            borderColor = Color.White.copy(alpha = 0.16f),
            buttonGradient = Brush.linearGradient(
                colors = listOf(Color(0xFF2563EB), Color(0xFF1D4ED8)),
            ),
        )
    }
}
